### PR TITLE
Feature/modal mount node

### PR DESCRIPTION
**Pull request by dmitrykrylov**
- [ ] Bug fix 
- [x] New feature 
- [ ] Chore 
- [ ] Breaking change 
- [x] There is an open issue which this change addresses
- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** document.
- [x] My commits follow the [Git Commit Guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- - [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Relates to #1274
Should I add any unit tests? I'm not sure how to better test this feature

### CODE REVIEW

File: src/Modal.js [hunk 1]
```diff
@@ -54,6 +54,7 @@ const propTypes = {
     PropTypes.string,
     PropTypes.func,
   ]),
+  mountNode: PropTypes.any,
```
Comment (TheSharpieOne) [inline src/Modal.js]: We should probably limit this like we do for the `target`s in tooltip/popover and allow for a DOMElement, function (which will return a DOMElement), or a string (which can be used to find a DOMElement).

There is a util which handles all of those cases, returning the DOMElement if found.

Probably should call it `container` to match similar functionality/props elsewhere.

File: src/Modal.js [hunk 2]
```diff
@@ -235,6 +236,8 @@ class Modal extends React.Component {
   }
 
   init() {
+    const mountNode = this.props.mountNode || document.body;
```
Comment (TheSharpieOne) [inline src/Modal.js]: could probably use `defaultProps` to default it to `document.body`

Comment (dmitrykrylov) [reply src/Modal.js]: Look at the my first commit. I used `defaultProps` but when I run `npm start` it threw an error where `document` was `undefined`.

Comment (dmitrykrylov) [reply src/Modal.js]: Probably webpack-dev-server works in kind of SSR mode?

Comment (TheSharpieOne) [reply src/Modal.js]: Yeah, I suppose that will happen on the server when document is not defined. What you have here should be fine then.

**Comment by virgofx** [comment]
This won't work with the forthcoming changes as well on server side. Everything in the init() will need to move to client side rendering as this breaks when {open} = true on SSR. Please do not merge this.

**Comment by iamandrewluca** [comment]
Fixed in #1817

**Pull request closed without merging**